### PR TITLE
fix: user model - storage is under state key not env

### PR DESCRIPTION
**Pull request by danielgent**
This should fix the issue I'm having with the delete users route :crossed_fingers:

### CODE REVIEW

File: src/models/User.ts [hunk 1]
```diff
@@ -294,7 +294,7 @@ export const userRouter = router({
       return profile;
     }),
   delete: publicProcedure.mutation(async ({ ctx }) => {
-    const profile = await getProfile(ctx.env.storage);
+    const profile = await getProfile(ctx.state.storage);
```
Comment (danielgent) [inline src/models/User.ts:297]: I'll try this same fix on my other PR and see if that solves the issue :+1:

File: test/models/User.spec.ts [hunk 2]
```diff
@@ -321,6 +321,41 @@ describe("User", () => {
...
+              profile = JSON.parse(value);
+          }
+          return;
+        },
+        delete: async () => {},
+      });
+
+      await caller.delete();
+
+      // what is even happening here? looks like I'm testing a mock
```
Comment (danielgent) [inline test/models/User.spec.ts:355]: All the above code is just guessing. Doesn't seem like I'm testing anything

Comment (markusahlstrand) [reply test/models/User.spec.ts:355]: So the get, put and delete are the mock for the storage. So what you need to validate is that the delete method on the storage is called.
If the get method returns null the delete function should throw a NotFound error as you're attempting to delete a non-existing user.

**Review by markusahlstrand — APPROVED** [review]
This works, but we should validate that the delete function is called on the storage. Or.. even the deleteAll, right?

**Comment by danielgent** [comment]
We've got this working now with other PRs.

**Pull request closed without merging**